File: Easy/Two Sum/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        paired_index= {}
        for i, num in enumerate(nums):
            if target - num in paired_index:
                return [i, paired_index[target-num]]
            paired_index[num] = i

#QED
#Problem 1 (Two Sum) - Jason Balayev (python)

File: Easy/Two Sum/TwoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> pairedIndex = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (pairedIndex.containsKey(target - num)) {
                return new int[] {i, pairedIndex.get(target - num)};
            }
            pairedIndex.put(num, i);
        }
        return new int[] {};
    }
}

//QED
//Problem 1 (Two Sum) - Jason Balayev (java)